lib/facter/apt_updates.rb

```diff
@@ -1,16 +1,32 @@
 apt_package_updates = nil
 Facter.add("apt_has_updates") do
   confine :osfamily => 'Debian'
-  if File.executable?("/usr/lib/update-notifier/apt-check")
-    apt_check_result = Facter::Util::Resolution.exec('/usr/lib/update-notifier/apt-check 2>&1')
-    if not apt_check_result.nil? and apt_check_result =~ /^\d+;\d+$/
-      apt_package_updates = apt_check_result.split(';')
+  if File.executable?("/usr/bin/apt-get")
+    apt_get_result = Facter::Util::Resolution.exec('/usr/bin/apt-get -s upgrade 2>&1')
+    if not apt_get_result.nil?
+      apt_package_updates = [[], []]
+      apt_get_result.each_line do |line|
+        if line =~ /^Inst\s/
+          package = line.gsub(/^Inst\s([^\s]+)\s.*/, '\1').strip
+          apt_package_updates[0].push(package)
+          security_matches = [
+            / Debian[^\s]+-updates /,
+            / Debian-Security:/,
+            / Ubuntu[^\s]+-security /,
+            / gNewSense[^\s]+-security /
+          ]
```

Comment:
I'm concerned about `security_matches` not working correctly on all target distributions such as Linux Mint etc.